### PR TITLE
Update all habits page to return default, custom habit of user and custom of his friends with status REQUESTED

### PR DESCRIPTION
**Pull request by HelenSotnik**
# GreenCity PR
[[All Habits] All Habits page shows all custom habits created by user's friends. Only custom habits REQUESTED to assign by friends should be shown instead of all. #6666](https://github.com/ita-social-projects/GreenCity/issues/6666)

## Summary Of Changes :fire:
- Updated HabitTranslationRepo, updated queries and methods names, docs for methods due to new meaning.

- Deleted few methods that are not used in code : findByNameAndLanguageCode, findHabitTranslationsByUserAndAcquiredStatus, findAllByLanguageCode; 

- Added new method findAllHabitIdsByUserIdAndStatusIsRequested() that returns  List<Long> of all habit ids REQUESTED by friends of current user.

- Updated implementations of methods in HabitServiceImpl: getAllHabitsByLanguageCode() and getAllByDifferentParameters();

- Updated tests in HabitServiceImplTest class for methods in HabitServiceImpl;

## Deleted
Methods in HabitTranslationRepo that are not used in code : findByNameAndLanguageCode(), findHabitTranslationsByUserAndAcquiredStatus(), findAllByLanguageCode(); 

# PR Checklist Forms
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 1]
```diff
@@ -117,13 +76,14 @@ List<HabitTranslation> findHabitTranslationsByUserAndAcquiredStatus(@Param("user
...
+        + "WHERE ((h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId))  "
+        + "OR (h.isCustomHabit = false)) "
+        + "AND t.id IN "
         + "(SELECT tt.tag FROM TagTranslation AS tt "
         + "WHERE lower(tt.name) IN (:tags))) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByTagsAndLanguageCodeAndForAvailableUsersIfIsCustomHabitTrue(Pageable pageable,
-        List<String> tags, String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findAllByTagsAndLanguageCodeAndByUserIdAndRequestedStatus(Pageable pageable,
+        List<String> tags, String languageCode, List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]:
```suggestion
        List<String> tags, String languageCode, List<Long> requestedCustomHabitIds, Long userId);
```

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 2]
```diff
@@ -170,33 +133,39 @@ Page<HabitTranslation> findAllByDifferentParametersIsCustomHabitFalse(Pageable p
...
+        + "WHERE (h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) "
+        + "AND h.complexity IN (:complexities) AND t.id IN "
         + "(SELECT tt.tag FROM TagTranslation AS tt "
         + "WHERE lower(tt.name) IN (:tags))) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByDifferentParametersIsCustomHabitTrue(Pageable pageable, List<String> tags,
-        Optional<List<Integer>> complexities, String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findCustomHabitsByDifferentParametersByUserIdAndStatusRequested(Pageable pageable,
+        List<String> tags, Optional<List<Integer>> complexities, String languageCode,
+        List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]:
```suggestion
        List<Long> requestedCustomHabitIds, Long userId);
```

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 3]
```diff
@@ -170,33 +133,39 @@ Page<HabitTranslation> findAllByDifferentParametersIsCustomHabitFalse(Pageable p
...
         + "(SELECT l FROM Language AS l WHERE l.code = :languageCode) "
         + "AND ht.habit IN "
         + "(SELECT h FROM Habit AS h "
-        + "WHERE h.isCustomHabit = true AND h.userId IN (:availableUsersIds)) "
+        + "WHERE (h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId))) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByIsCustomHabitTrueAndLanguageCode(Pageable pageable,
-        String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findCustomHabitsByLanguageCodeAndByUserIdAndStatusRequested(Pageable pageable,
+        String languageCode, List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]:
```suggestion
        String languageCode, List<Long> requestedCustomHabitIds, Long userId);
```

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 4]
```diff
@@ -218,36 +187,47 @@ Page<HabitTranslation> findAllByIsCustomHabitTrueAndLanguageCode(Pageable pageab
...
-        + "WHERE (( h.isCustomHabit = true AND h.userId IN (:availableUsersIds )) OR (h.isCustomHabit = false)) "
+        + "WHERE ((h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) "
+        + "OR (h.isCustomHabit = false)) "
         + "AND h.complexity IN (:complexities)) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByComplexityAndLanguageCodeAndForAvailableUsersIfIsCustomHabit(Pageable pageable,
-        Optional<List<Integer>> complexities,
-        String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findAllByComplexityAndLanguageCodeAndUserIdAndStatusRequested(Pageable pageable,
+        Optional<List<Integer>> complexities, String languageCode, List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]: same

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 5]
```diff
@@ -256,12 +236,12 @@ Page<HabitTranslation> findAllByComplexityAndLanguageCodeAndForAvailableUsersIfI
...
         + "JOIN h.tags AS t "
-        + "WHERE h.isCustomHabit = true AND h.userId IN (:availableUsersIds) AND t.id IN "
+        + "WHERE (h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) AND t.id IN "
         + "(SELECT tt.tag FROM TagTranslation AS tt "
         + "WHERE lower(tt.name) IN (:tags))) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByTagsAndIsCustomHabitTrueAndLanguageCode(Pageable pageable, List<String> tags,
-        String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findCustomHabitsByTagsAndLanguageCodeAndByUserIdAndStatusRequested(Pageable pageable,
+        List<String> tags, String languageCode, List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]: same

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 6]
```diff
@@ -305,35 +290,42 @@ Page<HabitTranslation> findAllByTagsAndIsCustomHabitFalseAndLanguageCode(Pageabl
...
+        + "WHERE ((h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) "
+        + "OR (h.isCustomHabit = false)) "
         + "AND h.complexity IN (:complexities) AND  t.id IN "
         + "(SELECT tt.tag FROM TagTranslation AS tt "
         + "WHERE lower(tt.name) IN (:tags))) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByTagsAndComplexityAndLanguageCodeForAvailableUsersIfIsCustomTrue(Pageable pageable,
-        List<String> tags, Optional<List<Integer>> complexities, String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findAllByTagsAndComplexityAndLanguageCodeAndByUserIdAndStatusRequested(Pageable pageable,
+        List<String> tags, Optional<List<Integer>> complexities, String languageCode, List<Long> habitIdsRequested,
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]: same

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 7]
```diff
@@ -305,35 +290,42 @@ Page<HabitTranslation> findAllByTagsAndIsCustomHabitFalseAndLanguageCode(Pageabl
...
         + "AND ht.habit IN "
         + "(SELECT h FROM Habit AS h "
-        + "WHERE h.isCustomHabit = true AND h.userId IN (:availableUsersIds) "
+        + "WHERE (h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) "
         + "AND h.complexity IN (:complexities)) "
         + "ORDER BY ht.habit.id DESC")
-    Page<HabitTranslation> findAllByIsCustomHabitTrueAndComplexityAndLanguageCode(Pageable pageable,
-        Optional<List<Integer>> complexities, String languageCode, List<Long> availableUsersIds);
+    Page<HabitTranslation> findCustomHabitsByComplexityAndLanguageCodeAndUserIdAndStatusRequested(Pageable pageable,
+        Optional<List<Integer>> complexities, String languageCode, List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]: same

File: dao/src/main/java/greencity/repository/HabitTranslationRepo.java [hunk 8]
```diff
@@ -364,4 +356,29 @@ Page<HabitTranslation> findAllByIsCustomHabitFalseAndComplexityAndLanguageCode(P
...
+    @Query("SELECT DISTINCT ht FROM HabitTranslation AS ht "
+        + "WHERE ht.language = "
+        + "(SELECT l FROM Language AS l WHERE l.code = :language) "
+        + "AND ht.habit IN "
+        + "(SELECT h FROM Habit AS h "
+        + "WHERE (h.isCustomHabit = true AND (h.id IN (:habitIdsRequested) OR h.userId = :userId)) "
+        + "OR h.isCustomHabit = false) "
+        + "ORDER BY ht.habit.id DESC")
+    Page<HabitTranslation> findAllByLanguageCodeAndHabitAssignIdsRequestedAndUserId(Pageable pageable, String language,
+        List<Long> habitIdsRequested, Long userId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/HabitTranslationRepo.java]: same

**Review by VadymMakitra — COMMENTED** [review]
Please check comment

**Review by VadymMakitra — APPROVED** [review]

**Review by Maksym-Lenets — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Pull request merged**